dotnet/SimpleSauce.Test/SauceSessionTests.cs

```diff
@@ -1,36 +1,76 @@
 using FluentAssertions;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
 using Moq;
+using Newtonsoft.Json.Linq;
+using OpenQA.Selenium.Edge;
 using SimpleSauce;
-[assembly: Parallelize(Workers = 100, Scope = ExecutionScope.MethodLevel)]
 
 namespace SimpleSauceTests
 {
     [TestClass]
-    public class SauceSessionTests
+    public class SauceSessionTests : BaseTest
     {
         [TestMethod]
-        public void ShouldTakeSauceOptions()
+        public void SauceSession_OptionsPassedIn_SetsConcreteDriver()
         {
-            SauceOptions options = new SauceOptions();
-            var session = new SauceSession(options);
-            session.Should().NotBeNull();
+            SauceOptions = new SauceOptions();
+            SauceSession = new SauceSession(SauceOptions);
+            SauceSession.DriverImplementation.Should().BeOfType(typeof(SauceDriver));
+        }
+        [TestMethod]
+        public void SauceSession_NoConstructorParam_OptionsInitialized()
+        {
+            SauceSession = new SauceSession();
+            Assert.IsNotNull(SauceSession.Options);
         }
         [TestMethod]
         public void GetDataCenter_Default_IsWest()
         {
-            var session = new SauceSession();
-            session.DataCenter.Should().BeEquivalentTo(DataCenter.UsWest);
+            SauceSession = new SauceSession();
+            SauceSession.DataCenter.Should().BeEquivalentTo(DataCenter.UsWest);
         }
         [TestMethod]
         public void Start_Default_IsChrome()
         {
-            var dummyManager = new Mock<IRemoteDriver>();
-            var session = new SauceSession(dummyManager.Object);
+            var dummyManager = new Mock<ISauceRemoteDriver>();
+            SauceSession = new SauceSession(dummyManager.Object);
+
+            SauceSession.Start();
+
+            SauceSession.ChromeOptions.Should().NotBeNull();
+        }
+        [TestMethod]
+        public void Start_WithEdge_SetsEdgeBrowser()
+        {
+            SauceOptions = new SauceOptions();
+            SauceOptions.WithEdge();
+            var dummyManager = new Mock<ISauceRemoteDriver>();
+            SauceSession = new SauceSession(SauceOptions, dummyManager.Object);
 
-            session.Start();
+            SauceSession.Start();
 
-            session.ChromeOptions.Should().NotBeNull();
+            SauceSession.Options.EdgeOptions.Should().NotBeNull();
         }
+        //[TestMethod]
+        //public void Start_WithEdge15_SetsCorrectOptions()
+        //{
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/app/SauceLabs/simple_sauce/pullRequest?prid=4510198)